libass/ass_coretext.c

```diff
@@ -129,21 +129,30 @@ static char *get_name(CTFontDescriptorRef fontd, CFStringRef attr)
 static bool fill_family_name(CTFontDescriptorRef fontd,
                              ASS_FontProviderMetaData *info)
 {
-    if (info->n_family)
-        return true;
-
     char *family_name = get_name(fontd, kCTFontFamilyNameAttribute);
     if (!family_name)
         return false;
 
-    info->families = malloc(sizeof(char *));
-    if (!info->families) {
-        free(family_name);
-        return false;
+    info->extended_family = family_name;
```

Comment:
Not sure if that can actually happen, but the `if (info->n_family) return true` guard suggests `fill_family_name` may be called more than once for the same `ASS_FontProviderMetaData` object. Furthermore, iiuc, the result from `get_name` must be `free`d.
If that's the case, won't the `info->extended_family` assignment leak memory on repeated calls?

Reply:
The `n->family` guard is for adding a family if `get_font_info` has failed to find one (since #491/https://github.com/libass/libass/pull/491/commits/96caf3e15c1e945607b2eab6517faf197e93c6e7), and the guard is still present, just moved below because now this function also unconditionally adds an `extended_family`.

`extended_family` is freed in `ass_font_provider_free_fontinfo`.

This function is only called once per font from `get_font_info_ct` from `process_descriptors`. I may have forgotten to add a `free` in `process_descriptors` though.

Reply:
Added a `free` in `process_descriptors`, thanks.